gateway-service-definitions/src/main/resources/services/hbaseui/2.1.0/rewrite.xml

```diff
@@ -0,0 +1,222 @@
+  </rule>
+  <rule dir="OUT" name="HBASEUI/hbase/outbound/master/children" pattern="/procedures.jsp?{**}">
+    <rewrite template="{$frontend[url]}/hbase/webui/master/procedures.jsp?host={$inboundurl[host]}?port={$inboundurl[port]}?{**}"/>
+  </rule>
+  <rule dir="OUT" name="HBASEUI/hbase/outbound/master/children" pattern="/processMaster.jsp?{**}">
+    <rewrite template="{$frontend[url]}/hbase/webui/master/processMaster.jsp?host={$inboundurl[host]}?port={$inboundurl[port]}?{**}"/>
+  </rule>
+  <rule dir="OUT" name="HBASEUI/hbase/outbound/master/children" pattern="/dump?{**}">
+    <rewrite template="{$frontend[url]}/hbase/webui/master/dump?host={$inboundurl[host]}?port={$inboundurl[port]}?{**}"/>
+  </rule>
+  <rule dir="OUT" name="HBASEUI/hbase/outbound/master/children" pattern="/jmx?{**}">
+    <rewrite template="{$frontend[url]}/hbase/webui/master/jmx?host={$inboundurl[host]}?port={$inboundurl[port]}?{**}"/>
+  </rule>
+  <rule dir="OUT" name="HBASEUI/hbase/outbound/master/children" pattern="/prof?{**}">
+    <rewrite template="{$frontend[url]}/hbase/webui/master/prof?host={$inboundurl[host]}?port={$inboundurl[port]}?{**}"/>
+  </rule>
+
+  <!-- RegionServer UI proxying -->
+  <rule dir="IN" name="HBASEUI/hbase/inbound/regionserver/root/qualified" pattern="*://*:*/**/hbase/webui/regionserver?{host}?port}">
+    <rewrite template="{$serviceScheme[HBASEUI]}://{host}:{port}/rs-status"/>
+  </rule>
+  <rule dir="IN" name="HBASEUI/hbase/inbound/regionserver/all_children" pattern="*://*:*/**/hbase/webui/regionserver/{**}?{host}?{port}?{**}">
+    <rewrite template="{$serviceScheme[HBASEUI]}://{host}:{port}/{**}?{**}"/>
+  </rule>
+  <rule dir="OUT" name="HBASEUI/hbase/outbound/regionserver/children" pattern="/rs-status?{**}">
+    <rewrite template="{$frontend[url]}/hbase/webui/regionserver/rs-status?host={$inboundurl[host]}?port={$inboundurl[port]}?{**}"/>
+  </rule>
+  <rule dir="OUT" name="HBASEUI/hbase/outbound/regionserver/children" pattern="/processRS.jsp">
+    <rewrite template="{$frontend[url]}/hbase/webui/regionserver/processRS.jsp?host={$inboundurl[host]}?port={$inboundurl[port]}"/>
+  </rule>
+  <rule dir="OUT" name="HBASEUI/hbase/outbound/regionserver/children" pattern="/conf">
+    <rewrite template="{$frontend[url]}/hbase/webui/regionserver/conf?host={$inboundurl[host]}?port={$inboundurl[port]}"/>
+  </rule>
+  <rule dir="OUT" name="HBASEUI/hbase/outbound/regionserver/children" pattern="/tablesDetailed.jsp">
+    <rewrite template="{$frontend[url]}/hbase/webui/regionserver/tablesDetailed.jsp?host={$inboundurl[host]}?port={$inboundurl[port]}"/>
+  </rule>
+  <rule dir="OUT" name="HBASEUI/hbase/outbound/regionserver/children" pattern="/procedures.jsp">
+    <rewrite template="{$frontend[url]}/hbase/webui/regionserver/procedures.jsp?host={$inboundurl[host]}?port={$inboundurl[port]}"/>
+  </rule>
+  <rule dir="OUT" name="HBASEUI/hbase/outbound/regionserver/children" pattern="/dump">
```

Comment:
So `/conf`, `/dump`, `/jmx`, and `/prof` (there might be others) all match both the regionserver and master handling. The redirect urls are written from a RS url back to a master url. It looks like the right data (since the host/port) is on the end - so it might make sense to make these rules not master/regionserver specific?

Starting from:

https://KNOX_HOST:8443/gateway/cluster_1/hbase/webui/regionserver/rs-status?host=HBASE_REGIONSERVER_HOST&port=22102

and clicking "Metrics Dump", I end up with the url:

https://KNOX_HOST:8443/gateway/cluster_1/hbase/webui/master/jmx?host=HBASE_REGIONSERVER_HOST&port=22102

The data coming back starts with 
```
{"beans":[{"name":"Hadoop:service=HBase,name=RegionServer,
```

so it looks like it is pulling the JMX metrics from the RegionServer, but the url is definitely `/hbase/webui/master/jmx`

Reply:
> So /conf, /dump, /jmx, and /prof (there might be others) all match both the regionserver and master handling. The redirect urls are written from a RS url back to a master url. It looks like the right data (since the host/port) is on the end - so it might make sense to make these rules not master/regionserver specific?

Yeah, these should just be non-specific to regionserver/master at this point. Pretty sure there aren't problems in practice (will double-check) but duplicity will be confusing (in a time where I'm trying to make this less confusing).